docs_src/microservices/application/AdvancedTopics.md

```diff
@@ -16,61 +16,61 @@ It is not uncommon to require your own custom REST endpoints when building an Ap
+      writer.Write([]byte("hello"))
+      writer.WriteHeader(200)
+    }
+
+    service := pkg.NewAppService(serviceKey)
+    service.AddCustomRoute("/myroute", service.Authenticated, myHandler, "GET")
+    ```
+
+Under the hood, this simply adds the provided route, handler, and method to the gorilla `mux.Router` used in the SDK. For more information on `gorilla mux` you can check out the github repo [here](https://github.com/gorilla/mux).
 You can access the `interfaces.ApplicationService` API for resources such as the logging client by pulling it from the context as shown above -- this is useful for when your routes might not be defined in your `main.go`  where you have access to the ``interfaces.ApplicationService`` instance.
 
 ### Target Type
 
-The target type is the object type of the incoming data that is sent to the first function in the function pipeline. By default this is an EdgeX `dtos.Event` since typical usage is receiving `Events` from the EdgeX MessageBus. 
+The target type is the object type of the incoming data that is sent to the first function in the function pipeline. By default this is an EdgeX `dtos.Event` since typical usage is receiving `Events` from the EdgeX MessageBus.
 
 There are scenarios where the incoming data is not an EdgeX `Event`. One example scenario is two application services are chained via the EdgeX MessageBus. The output of the first service is inference data from analyzing the original `Event`data, and published back to the EdgeX MessageBus. The second service needs to be able to let the SDK know the target type of the input data it is expecting.
 
 For usages where the incoming data is not `events`, the `TargetType` of the expected incoming data can be set when the `ApplicationService` instance is created using the `NewAppServiceWithTargetType()` factory function.
 
 !!! example "Example - Set and use custom Target Type"
 
-    ``` go    
-    type Person struct {    
-      FirstName string `json:"first_name"`    
-      LastName  string `json:"last_name"`    
-    }    
-        
-    service := pkg.NewAppServiceWithTargetType(serviceKey, &Person{})    
-    ```    
-    
+    ``` go
+    type Person struct {
+      FirstName string `json:"first_name"`
+      LastName  string `json:"last_name"`
+    }
+
+    service := pkg.NewAppServiceWithTargetType(serviceKey, &Person{})
+    ```
+
     `TargetType` must be set to a pointer to an instance of your target type such as `&Person{}` . The first function in your function pipeline will be passed an instance of your target type, not a pointer to it. In the example above, the first function in the pipeline would start something like:
-    
-    ``` go    
-    func MyPersonFunction(ctx interfaces.AppFunctionContext, data interface{}) (bool, interface{}) {    
-    
+
+    ``` go
+    func MyPersonFunction(ctx interfaces.AppFunctionContext, data interface{}) (bool, interface{}) {
+
       ctx.LoggingClient().Debug("MyPersonFunction executing")
-    
```

Comment:
Please turn off your white space remover in VS Code. It is too aggressive and remove tabs required for proper rendering in Github pages. Mkdocs will remove any unneeded white space when it renders the pages,

Ones here are not an issue, but muddy up this PR with unneeded changes.